File: 240116/줄 세우기/line-up-students.py
n = int(input())

class Student:

    def __init__(self,ki,he,num):
        self.ki = ki 
        self.he = he
        self.num = num

peo = []

for i in range(1,n+1):
    ki , he = map(int,input().split())
    peo.append(Student(ki,he,i))
peo.sort(key = lambda x: (-x.ki,-x.he,-x.num))
for i in range(n):
    print(peo[i].ki,peo[i].he,peo[i].num)

File: 240116/줄 세우기/line-up-students.java
import java.util.Scanner;
import java.util.Arrays;
class Student implements Comparable<Student>{
    int k,h,num;
    public Student(int k , int h, int num){
        this.k = k ;
        this.h = h ;
        this.num = num;
    }
    public int compareTo(Student student){
        if (this.k != student.k){
            return student.k - this.k;
        }
        if (this.h != student.h){
            return student.h -this.h;
        }
        return this.num - student.num;
    }
}
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Student[] peo = new Student[n];

        for(int i =0; i<n; i++){
            int k = sc.nextInt();
            int h = sc.nextInt();
            peo[i] = new Student(k,h,i+1);
        }
        Arrays.sort(peo);

        for(int i = 0; i<n; i++){
            System.out.println(peo[i].k+" "+peo[i].h+" "+peo[i].num);
        }
    }
}